pkg/build/build.go

```diff
@@ -53,37 +53,69 @@ import (
+		Paths []string
 	} `yaml:"trigger,omitempty"`
 
-	PreInstall    string `yaml:"pre-install,omitempty"`
-	PostInstall   string `yaml:"post-install,omitempty"`
-	PreDeinstall  string `yaml:"pre-deinstall,omitempty"`
+	// Optional: The script to run pre install. The script should contain the
+	// shebang interpreter.
+	PreInstall string `yaml:"pre-install,omitempty"`
+	// Optional: The script to run post install. The script should contain the
+	// shebang interpreter.
+	PostInstall string `yaml:"post-install,omitempty"`
+	// Optional: The script to run before uninstalling. The script should contain
+	// the shebang interpreter.
+	PreDeinstall string `yaml:"pre-deinstall,omitempty"`
+	// Optional: The script to run after uninstalling. The script should contain
+	// the shebang interpreter.
 	PostDeinstall string `yaml:"post-deinstall,omitempty"`
-	PreUpgrade    string `yaml:"pre-upgrade,omitempty"`
-	PostUpgrade   string `yaml:"post-upgrade,omitempty"`
+	// Optional: The script to run before upgrading. The script should contain
+	// the shebang interpreter.
+	PreUpgrade string `yaml:"pre-upgrade,omitempty"`
+	// Optional: The script to run after upgrading. The script should contain the
+	// shebang interpreter.
+	PostUpgrade string `yaml:"post-upgrade,omitempty"`
 }
 
 type PackageOption struct {
+	// Optional: Signify this package as a virtual package which does not provide
+	// any files, executables, librariries, etc... and is otherwise empty
 	NoProvides bool `yaml:"no-provides"`
-	NoDepends  bool `yaml:"no-depends"`
+	// Optional: Mark this package as a self contained package that does not
+	// depend on any other package
+	NoDepends bool `yaml:"no-depends"`
+	// Optional: Mark this package as not providing any executables
 	NoCommands bool `yaml:"no-commands"`
 }
 
 type Package struct {
-	Name               string        `yaml:"name"`
-	Version            string        `yaml:"version"`
-	Epoch              uint64        `yaml:"epoch"`
-	Description        string        `yaml:"description,omitempty"`
-	URL                string        `yaml:"url,omitempty"`
-	Commit             string        `yaml:"commit,omitempty"`
-	TargetArchitecture []string      `yaml:"target-architecture,omitempty"`
-	Copyright          []Copyright   `yaml:"copyright,omitempty"`
-	Dependencies       Dependencies  `yaml:"dependencies,omitempty"`
-	Options            PackageOption `yaml:"options,omitempty"`
-	Scriptlets         Scriptlets    `yaml:"scriptlets,omitempty"`
+	// The name of the package
+	Name string `yaml:"name"`
+	// The version of the package
+	Version string `yaml:"version"`
+	// The monotone increasing epoch of the package
+	Epoch uint64 `yaml:"epoch"`
+	// A human readable description of the package
+	Description string `yaml:"description,omitempty"`
+	// The URL to the package's homepage
+	URL string `yaml:"url,omitempty"`
+	// Optional: The git commit of the package
+	Commit string `yaml:"commit,omitempty"`
```

Comment:
nit: In case it helps to disambiguate, this is the commit of the package build config, not of the package's upstream source